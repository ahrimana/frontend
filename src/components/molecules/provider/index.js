import React from 'react';
import { connect } from 'react-redux';
import Paper from 'material-ui/Paper';
import ProviderIcon from 'material-ui/svg-icons/device/storage';
import actions from './actions/remove';
import styles from '../../atoms/icon/styles';


const mapStateToProps = (state) => ({
  theme: state.theme.theme,
});


export const Provider = React.createClass({
  propTypes: {
    name: React.PropTypes.string.isRequired,
    clusterId: React.PropTypes.string.isRequired,
    confirm: React.PropTypes.func.isRequired,
  },

  handleClose(event) {
    event.preventDefault();
    const clusterId = this.props.clusterId;
    const name = this.props.name;
    const id = `clusters/${clusterId}/providers/${name}`;
    this.props.confirm(id);
  },

  render() {
    return (
      <Paper style={styles.paper}>
        <div style={styles.close} onClick={this.handleClose}>x</div>
        <ProviderIcon color={styles.icon.color} style={styles.icon} />
        <div>{this.props.name}</div>
      </Paper>
    );
  },
});


export default connect(mapStateToProps, actions)(Provider);
