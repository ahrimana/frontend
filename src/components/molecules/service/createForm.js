import React from 'react';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import settingsActions from '../../layouts/layout/actions/settings';
import actions from './actions/create';

const CreateServiceForm = React.createClass({
  propTypes: {
    dispatch: React.PropTypes.func.isRequired,
  },

  getInitialState() {
    return {};
  },

  handleSubmit(event) {
    event.preventDefault();
    this.props.dispatch(actions.create(this.state.name));
    this.props.dispatch(settingsActions.close());
  },

  handleNameChange(event) {
    this.setState({ name: event.target.value });
  },

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <div>
          <TextField
            floatingLabelText="Name"
            onChange={this.handleNameChange}
            required
            autoFocus
          />
        </div>
        <div>
          <RaisedButton label="Create" type="submit" />
        </div>
      </form>
    );
  },
});


export default CreateServiceForm;
