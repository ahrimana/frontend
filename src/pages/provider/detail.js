import React from 'react'
import PropTypes from 'prop-types'
import { observer } from 'mobx-react'

// Components
import Button from '@material-ui/core/Button'
import Paper from '@material-ui/core/Paper'
import TextField from '@material-ui/core/TextField'
import Tooltip from '@material-ui/core/Tooltip'

import store from 'store'
import Template from 'templates/default'
import styles from './styles'


@observer
class ProviderDetail extends React.Component {
  state = {
    edit: null,
  }

  componentWillMount() {
    store.title.title = 'Provider Detail'
    store.provider.fetch(this.props.match.params.id)
  }

  handleClick = (item) => () => {
    this.setState({
      edit: item,
      [item]: store.provider.detail[item],
    })
  }

  handleCancel = (item) => () => {
    store.provider.detail[item] = this.state[item]
    this.setState({
      edit: null,
      [item]: null,
    })
  }

  handleName = (event) => {
    store.provider.detail.name = event.target.value
  }

  handleSubmit = async (event) => {
    event.preventDefault()
    const result = await store.provider.edit(
      store.provider.detail,
      { name: store.provider.detail.name },
    )
    if (result.status >= 400) {
      store.error.message = result.error
      store.error.open = true
    }
    this.setState({
      edit: null,
    })
  }

  render() {
    const name = this.state.edit === 'name'
      ? (
        <form onSubmit={this.handleSubmit}>
          <TextField
            value={store.provider.detail.name}
            onChange={this.handleName}
          />
          <Button
            variant="outlined"
            style={styles.button}
            color="primary"
            type="submit"
          >
            Save
          </Button>
          <Button
            variant="outlined"
            style={styles.button}
            color="secondary"
            onClick={this.handleCancel('name')}
          >
            Cancel
          </Button>
        </form>
      )
      : (
        <div
          onClick={this.handleClick('name')}
          role="button"
          onKeyDown={this.handleClick('name')}
          tabIndex={0}
        >
          <Tooltip placement="right" title="Click to edit">
            <h1 style={styles.name}>{store.provider.detail.name}</h1>
          </Tooltip>
        </div>
      )
    return (
      <Template>
        <div style={styles.root}>
          <Paper style={styles.paper}>
            {name}
            <div style={styles.id}>
              id: {store.provider.detail.id}
            </div>
          </Paper>
        </div>
      </Template>
    )
  }
}


ProviderDetail.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
}


export default ProviderDetail
