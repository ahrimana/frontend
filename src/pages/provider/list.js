import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import { withTheme } from '@material-ui/core/styles'
import { observer } from 'mobx-react'

// Components
import Avatar from '@material-ui/core/Avatar'
import Button from '@material-ui/core/Button'
import Checkbox from '@material-ui/core/Checkbox'
import Dialog from '@material-ui/core/Dialog'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogTitle from '@material-ui/core/DialogTitle'
import Fab from '@material-ui/core/Fab'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction'
import ListItemText from '@material-ui/core/ListItemText'
import TextField from '@material-ui/core/TextField'
import Tooltip from '@material-ui/core/Tooltip'

// Icons
import AddIcon from '@material-ui/icons/Add'

import store from 'store'
import Template from 'templates/default'
import ItemDelete from 'components/organisms/item-delete'
import styles from './styles'


@observer
class ProviderList extends React.Component {
  state = {
    delete: false,
    name: '',
    showMenu: false,
  }

  componentWillMount() {
    store.title.title = 'Provider List'
    store.provider.fetchAll()
  }

  async componentWillReceiveProps(nextProps) {
    const oldPage = Number(this.props.match.params.page || '0')
    const newPage = Number(nextProps.match.params.page || '0')
    if (oldPage !== newPage) {
      const result = await store.provider.fetchAll(newPage)
      if (result.status >= 400) {
        store.error.message = result.error
        store.error.open = true
      }
    }
  }

  handleDeleteOpen = () => {
    this.setState({ delete: true })
  }

  handleDeleteClose = () => {
    this.setState({ delete: false })
  }

  handleDelete = () => {
    const selected = store.provider.list.data.filter(provider => provider.selected)
    selected.forEach(provider => store.provider.remove(provider.id))
  }

  handleMenuOpen = () => {
    this.setState({ showMenu: true })
  }

  handleMenuClose = () => {
    this.setState({ showMenu: false })
  }

  handleProviderAdd = async (event) => {
    event.preventDefault()
    const result = await store.provider.create(this.state.name)
    if (result.status >= 400) {
      store.error.message = result.error
      store.error.open = true
    } else {
      this.handleMenuClose()
    }
  }

  handleProviderName = (event) => {
    this.setState({ name: event.target.value })
  }

  render() {
    const page = Number(this.props.match.params.page || '0')
    const previous = page !== 0
      ? (
        <Link
          to={page !== 1 ? `/providers/${page - 1}` : '/providers'}
          style={this.props.theme.overrides.noDecorationLink}
        >
          <Button variant="outlined">
            &lt;
          </Button>
        </Link>
      )
      : (
        <Button variant="outlined" disabled>
          &lt;
        </Button>
      )
    const next = page < store.provider.list.pages - 1
      ? (
        <Link
          to={`/providers/${page + 1}`}
          style={this.props.theme.overrides.noDecorationLink}
        >
          <Button variant="outlined">
            &gt;
          </Button>
        </Link>
      )
      : (
        <Button variant="outlined" disabled>
          &gt;
        </Button>
      )
    const providers = store.provider.list.data.map(provider => (
      <List style={styles.item} key={provider.id}>
        <ListItem dense button>
          <ListItemText primary={provider.name} />
          <ListItemSecondaryAction>
            <Link
              to={`/provider/${provider.id}`}
              style={this.props.theme.overrides.noDecorationLink}
            >
              <Button
                style={styles.detail}
                variant="outlined"
                color="primary"
              >
                Details
              </Button>
            </Link>
            <Checkbox
              onChange={() => { provider.selected = !provider.selected }}
              data-id={`checkbox-${provider.id}`}
              checked={provider.selected}
            />
          </ListItemSecondaryAction>
        </ListItem>
      </List>
    ))
    const selected = store.provider.list.data.filter(provider => provider.selected)
    return (
      <Template>
        <div style={styles.actions}>
          <Tooltip title="Add new task" placement="right">
            <Fab
              color="primary"
              onClick={this.handleMenuOpen}
              data-id="add-task"
              style={styles.add}
            >
              <AddIcon />
            </Fab>
          </Tooltip>
          <Dialog
            open={this.state.showMenu}
            onClose={this.handleMenuClose}
          >
            <DialogTitle>Add new provider</DialogTitle>
            <DialogContent>
              <TextField
                label="provider name"
                style={styles.text}
                onChange={this.handleProviderName}
                data-id="name"
                autoFocus
              />
            </DialogContent>
            <DialogActions>
              <Button
                variant="contained"
                color="primary"
                type="submit"
                onClick={this.handleProviderAdd}
                data-id="create"
              >
                Create
              </Button>
              <Button
                onClick={this.handleMenuClose}
                variant="contained"
                color="secondary"
              >
                Cancel
              </Button>
            </DialogActions>
          </Dialog>
          <Button
            variant="contained"
            color="secondary"
            disabled={selected.length === 0}
            onClick={this.handleDeleteOpen}
            data-id="delete"
          >
            Delete
          </Button>
        </div>
        <ItemDelete
          onDelete={this.handleDelete}
          onClose={this.handleDeleteClose}
          open={this.state.delete}
          title="providers"
        />
        {providers}
        <div style={styles.center}>
          {previous}
          <Avatar style={styles.page}>{String(page)}</Avatar>
          {next}
        </div>
      </Template>
    )
  }
}


ProviderList.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      page: PropTypes.string,
    }).isRequired,
  }).isRequired,
  theme: PropTypes.shape().isRequired,
}


export default withTheme()(ProviderList)
