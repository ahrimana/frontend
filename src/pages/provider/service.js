import axios from 'axios'
import { API_ROOT, getCookie } from 'utils'


async function create(name) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.post(
    `${API_ROOT}/provider`,
    { name },
    {
      headers: {
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


async function edit(provider, data) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.patch(
    `${API_ROOT}/provider/${provider.id}`,
    data,
    {
      headers: {
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


async function fetchAll(page) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.get(
    `${API_ROOT}/provider`,
    {
      headers: {
        'X-Page': page,
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


async function fetch(id) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.get(
    `${API_ROOT}/provider/${id}`,
    {
      headers: { 'X-CSRF-TOKEN': csrf },
    },
  )
  return response.data
}


async function remove(id) {
  const csrf = getCookie('csrf_access_token')
  const response = await axios.delete(
    `${API_ROOT}/provider/${id}`,
    {
      headers: {
        'X-CSRF-TOKEN': csrf,
      },
    },
  )
  return response.data
}


export default {
  create,
  edit,
  fetch,
  fetchAll,
  remove,
}
