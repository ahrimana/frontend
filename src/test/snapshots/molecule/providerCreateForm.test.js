/* eslint no-undef: 0 */
/* eslint no-unused-expressions: 0 */
/* eslint no-sequences: 0 */
/* eslint quotes: 0 */

import React from 'react';
import { CreateProviderForm } from '../../../components/molecules/provider/createForm';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

describe("Create provider form(snapshot)", () => {
  const setup = () => {
    const props = {
      cluster: {
        id: "2311231",
      },
      plugins: [
        {
          properties: [],
          type: "DO",
        },
        {
          properties: [
            {
              name: "name",
              type: "string",
            },
          ],
          type: "SSH",
        },
      ],
      dispatch: jest.fn(),
    };
    const muiTheme = getMuiTheme();
    const context = {
      context: { muiTheme },
      childContextTypes: { muiTheme: React.PropTypes.object },
    };
    return shallow(<CreateProviderForm { ...props } />, { ...context });
  };
  it("Create provider form render form", () => {
    const component = setup();
    expect(toJson(component)).toMatchSnapshot();
  });
});
