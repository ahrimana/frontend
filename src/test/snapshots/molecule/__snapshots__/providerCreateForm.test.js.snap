exports[`Create provider form(snapshot) Create provider form render form 1`] = `
<form
  onSubmit={[Function]}>
  <h2>
    Create Provider
  </h2>
  <SelectField
    autoWidth={false}
    disabled={false}
    fullWidth={false}
    hintText="Select Provider Type"
    onChange={[Function]}>
    <MenuItem
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="DO"
      value={0} />
    <MenuItem
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="SSH"
      value={1} />
  </SelectField>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }>
    <RaisedButton
      disabled={false}
      fullWidth={false}
      label="Create"
      labelPosition="after"
      primary={false}
      secondary={false}
      type="submit" />
  </div>
</form>
`;
